### DISCUSSION TITLE
XMake + CLang comand line exceded length

### DISCUSSION THREAD

**Discussion by kridenberg** [xmake-io/xmake#5224] [Q&A] [answered]
I want to build a module with clang++ that consists of a huge amount of submodules. Literally, I have a primary interface unit which re-exports (export import :partition) all its partition interface units. I have nearly 350 partitions to re-export, and clang++ cannot handle such a large command-line input. Is there any way to bypass that? I tried to fill an issue, but I am away from my target machine, and cannot publish code (due to NDA), and creationg minimal example is, well, a lot of work (assuming that I have 350 paritions to re-export)

**Comment by waruqi** [comment]
xmake should already handle long-path for clang link command. It should work.

https://github.com/xmake-io/xmake/blob/2e844e2df4207900ea6b9f0bf9c6049b0cac1c7f/xmake/modules/core/tools/gcc.lua#L565

So I need more information (logs, xmake.lua, compile commands) to be able to locate, which command is failing. Compile, link, or somewhere else.

or you can debug source code.

**Comment by kridenberg** [comment]
@waruqi I still believe it is in the command line length. I have copy-pasted arguments of the failed command and used a response file to pass them to the clang, and everything compiled as it should. I will provide you with commands I ran to bootstrap and build the project, the output of the -v -D of those commands, .xmake folder and args.rs I used to finalise the build.

[info.zip](https://github.com/user-attachments/files/15874537/info.zip)

Additionally, I want to say that the source code is completely fine and built with CMake. I am really enthusiastic to the xmake, and want to use it, but that is a real block for now. This is the of the info I can provide you with.

**Reply by waruqi** [reply, marked as answer]
I've only dealt with the link command before, but it's hard to think that your compilation commands would be so long as well. You could try this patch.

https://github.com/xmake-io/xmake/pull/5226

```console
xmake update -s github:xmake-io/xmake#longcmd
```

**Reply by kridenberg** [reply]
Thanks. I will try it as soon as I am around my workstation.
Regarding the compilation command length, I belive that this is the problem because when modules are used, we need to specify each separate BMI explicitly, you cannot (as far as I knew) just specify BMI output directory and hope that a compiler will detect those files on its own (like with includes), and due to that, we will stuck with those huge command lines from now.

**Reply by kridenberg** [reply]
This patch helped. The code compiled as expected. Thanks.

**Reply by waruqi** [reply]
I have merged it.
